driver/network_driver.go

```diff
@@ -261,10 +291,12 @@ func (d NetworkDriver) CreateEndpoint(request *network.CreateEndpointRequest) (*
 	}
 
 	endpoint := api.NewWorkloadEndpoint()
-	endpoint.Metadata.Node = hostname
-	endpoint.Metadata.Orchestrator = d.orchestratorID
-	endpoint.Metadata.Workload = d.containerName
-	endpoint.Metadata.Name = request.EndpointID
+	endpoint.ObjectMeta.Namespace = hostname
```

Comment:
Yup, that makes sense, but I think we should solve that more generally rather than overload the use of `namespace` - the wep already has a `node` field, so we can just make that an option in `calicoctl` for example - `calicoctl get weps --node mynode`

Reply:
Yeah, that's sound good. But in this time calicoctl not support filter weps by node yet . And in data model I think it's a bit hard to implement. Because now we query workloads endpoint by query etcd with `prefix` (like /calico/v3/projectcalico.org/weps) and it will response all children with their data. So if we want use `calicoctl get weps --node somenode` that's means we have to **unmarshal** each data and filter in calicoctl with huge costs ( netio, cpu, memory ) if we have lots containers. However, if we set namespace with weps, it will store with `prefix/{NAMESPACE}` then etcd will help us to filter out what we need with less costs.

Reply:
take a look at how the naming for libnetwork weps is meant to be: https://github.com/projectcalico/libcalico-go/blob/master/lib/names/workloadendpoint.go#L32

The first bit of the name is the node, and so lookup by node should be easily done at the etcd layer rather than by iteration.

Reply:
But in etcd I found it store without node info like this
```
/calico/resources/v3/projectcalico.org/workloadendpoints/dev1/dev1-libnetwork-libnetwork-a7aba82dc07d2331001ed6d517e0403b6eba36865bc1f23d6c8601f79ec0287b

/calico/resources/v3/projectcalico.org/workloadendpoints/dev1/dev1-libnetwork-libnetwork-cda1849e824ab06255b4576df33749e5d86d1bc29cdde6f59f127cf753530806

/calico/resources/v3/projectcalico.org/workloadendpoints/slave/slave-libnetwork-libnetwork-98e3b4b92b03cc498ea86aa0ee7bcb1b192aebb15a2c881568a8abdd1a2eee9f
```

```dev1``` ```slave``` both set by namespace, not nodename

Reply:
I think in your example above the format is like this:

```
/calico/resources/v3/projectcalico.org/workloadendpoints/<namespace>/<node>-libnetwork-libnetwork-<ID>
```

etcdv3 allows for prefix queries, so you can do a query against this:

```
/calico/resources/v3/projectcalico.org/workloadendpoints/<namespace>/<node>
```

To get all workloads within that namespace on that node.

Reply:
I think we have a few options that come to mind:
- Use `libnetwork` as the namespace for everything.
- Use `libnetwork.<network_name>` as the namespace.
- Optionally, allow passing in a label to set the namespace.

For OpenStack, we just use `openstack` as the namespace, so just using `libnetwork` seems OK to me.

Reply:
I think ```network_name``` and ```node``` both important for filtering weps in huge system. hmmm,  ```libnetwork.<network_name>``` seems better since node name was part of weps' name.